monai/apps/auto3dseg/algorithms/templates/dints/scripts/train.py

```diff
@@ -0,0 +1,400 @@
+
+    with open(data_list_file_path) as f:
+        json_data = json.load(f)
+
+    list_train = []
+    list_valid = []
+    for item in json_data["training"]:
+        if item["fold"] == fold:
+            item.pop("fold", None)
+            list_valid.append(item)
+        else:
+            item.pop("fold", None)
+            list_train.append(item)
+
+    files = []
+    for _i in range(len(list_train)):
+        str_img = os.path.join(data_file_base_dir, list_train[_i]["image"])
+        str_seg = os.path.join(data_file_base_dir, list_train[_i]["label"])
+
+        if (not os.path.exists(str_img)) or (not os.path.exists(str_seg)):
+            continue
+
+        files.append({"image": str_img, "label": str_seg})
+
+    train_files = files
+    random.shuffle(train_files)
+
+    if torch.cuda.device_count() > 1:
+        train_files = partition_dataset(data=train_files, shuffle=True, num_partitions=world_size, even_divisible=True)[
+            dist.get_rank()
+        ]
+    print("train_files:", len(train_files))
+
+    files = []
+    for _i in range(len(list_valid)):
+        str_img = os.path.join(data_file_base_dir, list_valid[_i]["image"])
+        str_seg = os.path.join(data_file_base_dir, list_valid[_i]["label"])
+
+        if (not os.path.exists(str_img)) or (not os.path.exists(str_seg)):
+            continue
+
+        files.append({"image": str_img, "label": str_seg})
+
+    val_files = files
+
+    if torch.cuda.device_count() > 1:
+        if len(val_files) < world_size:
+            val_files = val_files * math.ceil(float(world_size) / float(len(val_files)))
+
+        val_files = partition_dataset(data=val_files, shuffle=False, num_partitions=world_size, even_divisible=False)[
+            dist.get_rank()
+        ]
+    print("val_files:", len(val_files))
+
+    if torch.cuda.device_count() >= 4:
+        train_ds = monai.data.CacheDataset(data=train_files, transform=train_transforms, cache_rate=1.0, num_workers=8)
+        val_ds = monai.data.CacheDataset(data=val_files, transform=val_transforms, cache_rate=1.0, num_workers=2)
+    else:
+        train_ds = monai.data.CacheDataset(
+            data=train_files,
+            transform=train_transforms,
+            cache_rate=float(torch.cuda.device_count()) / 4.0,
+            num_workers=8,
+        )
+        val_ds = monai.data.CacheDataset(
+            data=val_files, transform=val_transforms, cache_rate=float(torch.cuda.device_count()) / 4.0, num_workers=2
+        )
+
+    train_loader = DataLoader(train_ds, num_workers=8, batch_size=num_images_per_batch, shuffle=True)
+    val_loader = DataLoader(val_ds, num_workers=0, batch_size=1, shuffle=False)
+
+    device = torch.device(f"cuda:{dist.get_rank()}") if torch.cuda.device_count() > 1 else torch.device("cuda:0")
+    torch.cuda.set_device(device)
+
+    model = parser.get_parsed_content("network")
+    model = model.to(device)
+    model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
```

Comment:
Same as my comment in infer.py, SyncBatchNorm caused some issue when I trained with one GPU

Reply:
which model did you train?